### PR TITLE
feat: support symbolic indexing of a subset of the system

### PR DESCRIPTION
**Pull request by AayushSabharwal**
Draft while I add more tests

## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/solutions/ode_solutions.jl [hunk 1]
```diff
@@ -180,6 +184,58 @@ function SymbolicIndexingInterface.is_parameter_timeseries(::Type{S}) where {
...
+    T8,
+    T9,
+    T10,
+    T11,
+    T12,
+    T13,
+    T14,
+    T15,
+    T16} where {
+    T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16 <: SavedSubsystem}
```
Comment (ChrisRackauckas) [inline src/solutions/ode_solutions.jl:203]: this seems a bit fragile. Why not a trait function?

**Comment by ChrisRackauckas** [comment]
This should roll out to all solution types?

File: src/solutions/ode_solutions.jl (hunk 1 above)
Comment (AayushSabharwal) [reply src/solutions/ode_solutions.jl:203]: Yeah I guess that is a better option

**Comment by AayushSabharwal** [comment]
Yes. I'll add the rest.

**Pull request merged**